include/ocpp/v201/evse_manager.hpp

```diff
@@ -0,0 +1,68 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2020 - 2024 Pionix GmbH and Contributors to EVerest
+
+#pragma once
+
+#include <ocpp/common/utils.hpp>
+#include <ocpp/v201/evse.hpp>
+
+namespace ocpp {
+namespace v201 {
+
+class EvseOutOfRangeException : public std::exception {
+public:
+    explicit EvseOutOfRangeException(int32_t id) : msg{"Evse with id " + std::to_string(id) + " does not exist"} {
+    }
+
+    ~EvseOutOfRangeException() noexcept override = default;
+
+    const char* what() const noexcept override {
+        return msg.c_str();
+    }
+
+private:
+    std::string msg;
+};
+
+class EvseManagerInterface {
+private:
+    virtual std::vector<std::unique_ptr<EvseInterface>>& get_vector() = 0;
+    virtual const std::vector<std::unique_ptr<EvseInterface>>& get_vector() const = 0;
+
+public:
+    using EvseIterator = VectorOfUniquePtrIterator<EvseInterface>;
```

Comment:
Why is a const vector with const unique ptrs not possible?
`const std::vector<const std::unique_ptr<const EvseInterface>>` or something?